File: 2023/day-01/c/sol1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(){
    FILE* file = fopen("../in.txt", "r");
    char line[128];
    int n, sum = 0;
    while(n = fscanf(file, "%s", line) != EOF) {
        int fd = 0, ld = 0, first = 1;
        for(int i=0; i < strlen(line); i++)
            if(line[i] >= '0' && line[i] <= '9') {
                if(first) {
                    fd = line[i] - '0';
                    first = 0;
                }
                ld = line[i] - '0';
            }
        sum += fd*10 + ld;
        memset(line, 0, n);
    }
    printf("%d \n", sum);
    fclose(file);
}

File: 2023/day-01/go/sol1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	file, err := os.Open("./in.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	buf := make([]byte, 21*1024)
	reader := bufio.NewReader(file)
	n, err := reader.Read(buf)
	if err != nil {
		panic(err)
	}
	buf = buf[:n]

	var sum int
	for _, line := range strings.Split(string(buf), "\n") {
		if len(line) == 0 {
			continue
		}
		var fd, ld int
		first := true
		for _, chr := range line {
			chr := byte(chr)
			if chr >= '0' && chr <= '9' {
				if first {
					fd = int(chr - '0')
					first = false
				}
				ld = int(chr - '0')
			}
		}
		sum += fd*10 + ld
	}

	fmt.Printf("Sum is %v\n", sum)
}
